scrapy/utils/reqser.py

```diff
@@ -73,23 +73,22 @@ def request_from_dict(d, spider=None):
 def _find_method(obj, func):
```

Comment:
Maybe you could simplify this statement to something like this:

```python
def _find_method(obj, func):
    if obj and hasattr(func, '__func__'):
            members = inspect.getmembers(obj, predicate=inspect.ismethod)
            for name, obj_func in members:
                # We need to use __func__ to access the original
                # function object because instance method objects
                # are generated each time attribute is retrieved from
                # instance.
                #
                # Reference: The standard type hierarchy
                # https://docs.python.org/3/reference/datamodel.html
                if obj_func.__func__ is func.__func__:
                    return name
    raise ValueError("Function %s is not an instance method in: %s" % (func, obj))
```

Reply:
Good idea! Done here: https://github.com/scrapy/scrapy/pull/4756/commits/0524df866936506ae9438a5565fc4733fa5ba5b0